src/purchase/tests/test_fundraise_view.py

```diff
@@ -0,0 +1,219 @@
+            ):
+        if user is None:
+            user = self.user
+
+        self.client.force_authenticate(user)
+        return self.client.post(
+            "/api/fundraise/",
+            {
+                "post_id": post_id,
+                "recipient_user_id": user.id,
+                "goal_amount": goal_amount,
+                "goal_currency": goal_currency,
+            },
+        )
+
+    def _create_contribution(
+            self,
+            fundraise_id,
+            user,
+            amount = 100,
+            amount_currency = "RSC",
+            ):
+        self.client.force_authenticate(user)
+        return self.client.post(
+            f"/api/fundraise/{fundraise_id}/create_contribution/",
+            {
+                "amount": amount,
+                "amount_currency": amount_currency,
+            },
+        )
+    
+    def _get_contributions(self, fundraise_id):
+        return self.client.get(f"/api/fundraise/{fundraise_id}/contributions/")
+    
+    def _give_user_balance(self, user, amount):
+        DISTRIBUTION_CONTENT_TYPE = ContentType.objects.get(model="distribution")
+        Balance.objects.create(
+            amount=amount, user=user, content_type=DISTRIBUTION_CONTENT_TYPE
+        )
+
+
+    # Fundraise tests
+
+    def test_create_fundraise(self):
+        response = self._create_fundraise(self.post.id)
+
+        self.assertIsNotNone(response.data["id"])
+
+        # check fundraise
+        self.assertEqual(response.data["goal_amount"]["usd"], 100)
+        self.assertEqual(response.data["goal_currency"], "USD")
+        self.assertEqual(response.data["amount_raised"]["usd"], 0)
+        # check created_by
+        self.assertEqual(response.data["created_by"]["id"], self.user.id)
+        # check escrow
+        self.assertEqual(response.data["escrow"]["hold_type"], "FUNDRAISE")
+        self.assertEqual(float(response.data["escrow"]["amount_holding"]), 0.0)
+
+    def test_create_fundraise_no_post(self):
+        response = self._create_fundraise(9999)
+        self.assertEqual(response.status_code, 400)
+
+    def test_create_fundraise_not_preregistration(self):
+        self.client.force_authenticate(self.user)
+        post = create_post(created_by=self.user, document_type="DISCUSSION")
+        response = self._create_fundraise(post.id)
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_create_fundraise_already_exists(self):
+        response = self._create_fundraise(self.post.id)
+
+        self.assertIsNotNone(response.data["id"])
+
+        response = self._create_fundraise(self.post.id)
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_create_fundraise_not_moderator(self):
+        user = create_random_authenticated_user("fundraise_views")
+        self.client.force_authenticate(user)
+        response = self._create_fundraise(self.post.id, user=user)
+
+        self.assertEqual(response.status_code, 403)
+
+    # Contribution tests
```

Comment:
Good tests here. Can we create a couple more tests:
- One that ensures the status of a fundraise changes to "CLOSED" 
- One that ensure user cannot contribute if fundraise is "CLOSED"

Reply:
The fundraise_fulfilled test already checks that the status changes to `CLOSED`, but I added two more tests:
- Cannot contribute if fundraise is `CLOSED`
- Cannot contribute if fundraise is expired

Reply:
Awesome. TY